pages/index.js

```diff
@@ -47,12 +50,38 @@ export default function HomePage({ onSubmit }) {
 
         setWeather(mainTemp);
         setCondition(condition);
+        if (appUsers) {
+          const user = appUsers.find((user) => user._id === session?.user?.id);
+
+          if (user && user.friends) {
+            setAppUserFriendsList(user.friends);
+          } else {
+            return;
+          }
+        }
       } catch (error) {
         console.error(error);
       }
     }
-    fetchData();
-  }, [city]);
+
+    if (appUsers) {
+      fetchData();
+    }
+  }, [appUsers, session, city]);
+
+  if (!appUsers) return <div>Loading...</div>;
+
+  if (!activities) return <div>failed to load</div>;
+
+  if (activitiesIsLoading || status === "loading" || appUsersIsLoading)
+    return <div>loading...</div>;
+  if (!session) {
+    return (
+      <>
+        <LoginPage />
+      </>
```

Comment:
We don't need the fragment if we only return one thing. 

```suggestion
        <LoginPage />
```